### PR TITLE
Adding database vulnerability assessmen scan APIs

### PR DESCRIPTION
**Pull request by yaakoviyun**
Adding database vulnerability assessmen scan APIs

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json
Comment (anuchandy) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json]: The storage account should be referenced using the fully qualified ARM resource id instead of passing the url and access-key. You can then enable linked access check on it reliably to ensure sufficient permission are there to perform the operation. Since this is an ARM guideline, tagging Gaurav @ravbhatnagar

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json
Comment (anuchandy) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json]: Just confirming, is there any final response that contains the scan id? or it should always be retrieved using `DatabaseVulnerabilityAssessmentScans_ListByDatabase`?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json
Comment (anuchandy) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json]: If scan result was already converted, does this POST call return that existing result?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json
Comment (yaakoviyun) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json]: Currently to get the scan ID of a past scan, the user should use: DatabaseVulnerabilityAssessmentScans_ListByDatabase

If the user has a scan ID he can use DatabaseVulnerabilityAssessmentScans_Get

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json
Comment (yaakoviyun) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json]: yes

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json
Comment (yaakoviyun) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json]: We already got the approval from ARM to go this way before.
The intention here is that a user will supply a SAS key (not access key) with an expiration date and privileges he chooses to supply.
This way he is more "secure" as we have a SAS key that he controls on.

**Comment by anuchandy** [comment]
@yaakoviyun thanks for the response. @ravbhatnagar please sign-off since this is a new spec.

**Comment by anuchandy** [comment]
@yaakoviyun Could you share your MS alias, wanted to start an email thread but could't locate your name/alias.

**Comment by anuchandy** [comment]
@ravbhatnagar ping

**Comment by anuchandy** [comment]
@ravbhatnagar can you please take a look?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json]: Why is a scan not created (initiated) using PUT instead of POST?

**Comment by ravbhatnagar** [comment]
Just one comment around the use of POST instead of PUT.

**Comment by ravbhatnagar** [comment]
@yaakoviyun

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json
Comment (yaakoviyun) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json]: Put is traditionally used to create a resource. This API is for executing a scan (doing operation on a resource), so we thought POST will be a better match.

Comment (ravbhatnagar) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json]: @yaakoviyun - But this is not modeled as an action on a resource. Should it be modeled like - POST subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/vulnerabilityAssessments/{vulnerabilityAssessmentName}/initiateScan

Comment (yaakoviyun) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json]: OK Will make the change in the next API-version of this API.

**Comment by anuchandy** [comment]
@yaakoviyun can you please take a look at the one comment from Gaurav?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json
Comment (yaakoviyun) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json]: We decided to implement it for this api version

**Comment by yaakoviyun** [comment]
Ping.

Regards,
Yaakov

**Comment by yaakoviyun** [comment]
@ravbhatnagar can you take a look?

**Comment by ravbhatnagar** [comment]
Looks good!

File: specification/sql/resource-manager/readme.md
Comment (jaredmoo) [inline specification/sql/resource-manager/readme.md]: Undo whitespace

File: specification/sql/resource-manager/readme.md
Comment (jaredmoo) [inline specification/sql/resource-manager/readme.md]: Please add to both package-composite-v1 and -v2 sections, and package-pure-2017-10-preview

**Review by jaredmoo — CHANGES_REQUESTED** [review]

**Comment by anuchandy** [comment]
@yaakoviyun could you address comments from @jaredmoo?

**Comment by yaakoviyun** [comment]
Done.

Regards,
Yaakov

**Review by jaredmoo — APPROVED** [review]
Thanks for readme.md changes, since last month there is now a new composite-v3 section. Please add there as well. I am approving now because I am on vacation and might not look at your next commit :)

**Comment by yaakoviyun** [comment]
ping, can you merge?

**Comment by jaredmoo** [comment]
@yaakoviyun  There are merge conflicts again. Can you resolve? 

@ravbhatnagar it looks like you [signed off in comments](https://github.com/Azure/azure-rest-api-specs/pull/2418#issuecomment-371751834), can you approve?

**Comment by yaakoviyun** [comment]
ping...

**Review by anuchandy — APPROVED** [review]

**Pull request merged**
